api-tester/src/main/java/com/revenuecat/apitester/kotlin/PurchasesCommonAPI.kt

```diff
@@ -85,21 +86,32 @@ private class PurchasesCommonAPI {
         }
 
         val oldProductId = "old"
+        val replacementMode = GoogleReplacementMode.WITH_TIME_PRORATION
         val prorationMode = GoogleProrationMode.IMMEDIATE_WITH_TIME_PRORATION
         val isPersonalizedPrice = true
 
         val purchasePackageBuilder: PurchaseParams.Builder = PurchaseParams.Builder(activity, packageToPurchase)
         purchasePackageBuilder.oldProductId(oldProductId).googleProrationMode(prorationMode)
+        purchasePackageBuilder
+            .oldProductId(oldProductId)
+            .googleReplacementMode(replacementMode)
+            .googleProrationMode(prorationMode)
         val purchasePackageParams: PurchaseParams = purchasePackageBuilder.build()
         purchases.purchase(purchasePackageParams, purchaseCallback)
 
         val purchaseProductBuilder: PurchaseParams.Builder = PurchaseParams.Builder(activity, storeProduct)
-        purchaseProductBuilder.oldProductId(oldProductId).googleProrationMode(prorationMode)
+        purchaseProductBuilder
+            .oldProductId(oldProductId)
+            .googleReplacementMode(replacementMode)
+            .googleProrationMode(prorationMode)
         val purchaseProductParams: PurchaseParams = purchaseProductBuilder.build()
         purchases.purchase(purchaseProductParams, purchaseCallback)
 
         val purchaseOptionBuilder: PurchaseParams.Builder = PurchaseParams.Builder(activity, subscriptionOption)
-        purchaseOptionBuilder.oldProductId(oldProductId).googleProrationMode(prorationMode)
+        purchaseOptionBuilder
+            .oldProductId(oldProductId)
+            .googleReplacementMode(replacementMode)
+            .googleProrationMode(prorationMode)
```

Comment:
How come this is duplicated 3 times?

Reply:
It uses three different builders: for Package, Product and SubscriptionOption.